src/main/cpp/io/ShooterIO.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include "io/ShooterIO.h"
+
+void ShooterIO::Periodic() {
+    // Process all the inputs and outputs to/from high level software.
+    VOKC_CALL(ProcessIO());
+}
+
+void ShooterIO::SimulationPeriodic() {
+    // SimulationPeriodic
+}
+
+bool ShooterIO::ProcessIO() {
+    OKC_CHECK(sw_interface_ != nullptr);
+    OKC_CHECK(hw_interface_ != nullptr);
+
+    // Update the shooter motor config if needed
+    if (sw_interface_->update_shooter_config) {
+        OKC_CALL(UpdateShooterConfig());
+        sw_interface_->update_shooter_config = false;
+    }
+
+    // Update the trigger motor config if needed
+    if (sw_interface_->update_trigger_config) {
+        OKC_CALL(UpdateTriggerConfig());
+        sw_interface_->update_trigger_config = false;
+    }
+
+    // Set motor powers
+    OKC_CALL(SetMotorOutputs());
+
+    // Read sensors
+    OKC_CALL(ReadSensors());
```

Comment:
does this need to be the way we do IO classes now? Separating the output and input stuff into separate functions?

Reply:
It's definitely cleaner from a reading perspective. I think it's probably a good idea to make functions for the input and output stuff, and maybe even more functions for specific I/O operations if they are complicated enough